public/src/components/channelManagement/headerTests/headerTestVariantEditor.tsx

```diff
@@ -0,0 +1,286 @@
+  };
+
+  const labelSuffix = getLabelSuffix(deviceType);
+
+  const headingCopyLength = content.heading?.length ?? 0;
+  const subheadingCopyLength = content.subheading?.length ?? 0;
+
+  return (
+    <>
+      <Typography className={classes.sectionHeader} variant="h4">
+        {`Content${labelSuffix}`}
+      </Typography>
+
+      <div className={classes.contentContainer}>
+        <div>
+          <TextField
+            inputRef={register({ validate: templateValidator })}
+            error={errors.heading !== undefined}
+            helperText={errors.heading ? errors.heading.message : HEADING_DEFAULT_HELPER_TEXT}
+            onBlur={handleSubmit(onSubmit)}
+            name="heading"
+            label="Heading"
+            margin="normal"
+            variant="outlined"
+            disabled={!editMode}
+            fullWidth
+          />
+
+          {headingCopyLength > HEADING_COPY_RECOMMENDED_LENGTH && (
+            <VariantEditorCopyLengthWarning charLimit={HEADING_COPY_RECOMMENDED_LENGTH} />
+          )}
+        </div>
+      </div>
+
+      <div className={classes.contentContainer}>
+        <div>
+          <TextField
+            inputRef={register({ validate: templateValidator })}
+            error={errors.heading !== undefined}
+            helperText={errors.heading ? errors.heading.message : SUBHEADING_DEFAULT_HELPER_TEXT}
+            onBlur={handleSubmit(onSubmit)}
+            name="subheading"
+            label="Sub-heading"
+            margin="normal"
+            variant="outlined"
+            disabled={!editMode}
+            fullWidth
+          />
+
+          {subheadingCopyLength > SUBHEADING_COPY_RECOMMENDED_LENGTH && (
+            <VariantEditorCopyLengthWarning charLimit={SUBHEADING_COPY_RECOMMENDED_LENGTH} />
+          )}
+        </div>
+      </div>
+
+      <div className={classes.buttonsContainer}>
+        <Typography className={classes.sectionHeader} variant="h4">
+          {`Buttons${labelSuffix}`}
+        </Typography>
+
+        <HeaderTestVariantEditorCtasEditor
+          primaryCta={content.primaryCta}
+          secondaryCta={content.secondaryCta}
+          updatePrimaryCta={updatePrimaryCta}
+          updateSecondaryCta={updateSecondaryCta}
+          isDisabled={!editMode}
+          onValidationChange={onValidationChange}
+          supportSecondaryCta={true}
+        />
+      </div>
+    </>
+  );
+};
+
+interface HeaderTestVariantEditorProps {
+  variant: HeaderVariant;
+  onVariantChange: (updatedVariant: HeaderVariant) => void;
+  editMode: boolean;
+  onDelete: () => void;
+  onValidationChange: (isValid: boolean) => void;
+}
+
+const HeaderTestVariantEditor: React.FC<HeaderTestVariantEditorProps> = ({
+  variant,
+  editMode,
+  onValidationChange,
+  onVariantChange,
+}: HeaderTestVariantEditorProps) => {
+  const classes = useStyles();
+  const setValidationStatusForField = useValidation(onValidationChange);
+
+  const content: HeaderContent = variant.content || {
```

Comment:
the `content` field is mandatory in the model, so the `|| {...}` stuff can be removed

Reply:
Done